### ISSUE TITLE
Audiobook "progression" in publication info popup modal dialog (NaN Not A Number)

### ISSUE THREAD

**Issue by danielweck** [edrlab/thorium-reader#1349]
<img width="1276" alt="Screenshot 2021-01-29 at 10 54 01" src="https://user-images.githubusercontent.com/625201/106269216-22601900-6224-11eb-9750-7a4fd3617004.png">

**Comment by danielweck** [comment]
Compare with another audiobook (no "progression" info ... not sure where this comes from):

<img width="1276" alt="Screenshot 2021-01-29 at 10 56 00" src="https://user-images.githubusercontent.com/625201/106269287-3d328d80-6224-11eb-8dea-ef848dffbd6d.png">

**Comment by panaC** [comment]
https://bookserver.archive.org/group/openaudiobooks

**Comment by panaC** [comment]
i cannot found this audiobook with https://bookserver.archive.org/opensearch?q=Mind%20The%20Paint

**Comment by panaC** [comment]
<img width="812" alt="image" src="https://user-images.githubusercontent.com/3198865/132894243-45ba124f-6569-435f-ba15-53f0bd84a3a2.png">

I can't reproduce this issue

**Comment by panaC** [comment]
![Uploading image.png…]()

**Comment by danielweck** [comment]
I have just reproduced this from the develop branch :)

`mindthepaintgirl_1911_librivox.lpf`

<img width="1552" alt="Screenshot 2021-09-10 at 18 41 48" src="https://user-images.githubusercontent.com/625201/132895374-3d4e8392-acca-43d8-95e1-a8cf3a182127.png">

**Comment by danielweck** [comment]
I think this is division by zero when the duration metadata is missing.

**Comment by danielweck** [comment]
...mhmm, no. My assumption was incorrect, see:

`compilationpoemes_001_1007_librivox.lpf`

![Uploading Screenshot 2021-09-10 at 18.46.40.png…]()

**Comment by danielweck** [comment]
Also works with `multilingual_poetry_015_1007_librivox.lpf` which does not seem to include duration metadata (probably extracted at parsing stage, or obtained from audio navigator which returns total duration?)

<img width="1552" alt="Screenshot 2021-09-10 at 18 49 40" src="https://user-images.githubusercontent.com/625201/132896184-d8a8fe69-d873-4403-8404-2033faf1efa2.png">

**Comment by danielweck** [comment]
> `mindthepaintgirl_1911_librivox.lpf`

Ah yes, in the console:

```
[2]   audioPlaybackInfo: {
[2]     globalDuration: undefined,
[2]     globalProgression: undefined,
[2]     globalTime: undefined,
[2]     isPlaying: false,
[2]     localDuration: 4310.674286,
[2]     localProgression: 0.6504265135749112,
[2]     localTime: 2803.776847
[2]   },
```


...my guess is there is no duration on the reading order audio resources?

**Comment by danielweck** [comment]
> `mindthepaintgirl_1911_librivox.lpf`

`publication.json`

```json
{
  "@context": ["https://schema.org", "https://www.w3.org/ns/pub-context"],
  "dcterms:description": "Another Pinero play from the early 1900s. With a social message about the effects of stardom on the star as well as those that love them. \"I’m afraid there’s one thing finer than winning the woman you love and, when you’ve won her, being prepared to go through fire and water for her.\" \"What’s that?\" \"Having the courage to give her up\"\n - Summary by ToddHW\nCast list:\nViscount Farncombe: Tomas Peter\nColonel the Hon. Arthur Stidulph: alanmapstone\nBaron von Rettenmayer: Nemo\nCaptain Nicholas Jeyes: Campbell Schelp\nLionel Roper: Hamlet\nSam de Castro: Aaron White\nHerbert Fulkerson: Frédéric Surget\nStewart Heneage: KevinS\nGerald Grimwood: April 6090\nCarlton Smythe (Manager of the Pandora Theatre): James Thomas\nAlbert Palk (Actor at the Pandora): Sandra Schmit\nWilfrid Tavish (Actor at the Pandora): Philip Watson\nVincent Bland (A Musical Composer, attached to the Pandora): ToddHW\nMorris Cooling (Business Manager at the Pandora): Jim Locke\nLuigi (Maître d’hôtel at Catani’s Restaurant): Pier\nThe Hon. Mrs. Arthur Stidulph (Formerly, as Dolly Ensor, of the Pandora Theatre): TJ Burns\nLily Parradell (Of the Pandora): Foon\nJimmie Birch (Of the Pandora): Sonia\nGabrielle Kato (Of the Pandora): Pauline Latournerie\nEnid Moncrieff (Of the Pandora): Leanne Yau\nDaphne Dure (Of the Pandora): EmmaHatton\nNita Trevenna (Of the Pandora): Lex Hankins\nFlo Connify (One of Four Beauties Of the Pandora): Eva Davis\nSybil Dermott (One of Four Beauties Of the Pandora): Availle\nOlga Cook (One of Four Beauties Of the Pandora): Devorah Allen\nEvangeline Ventris (One of Four Beauties Of the Pandora): Elsie Selwyn\nMrs. Upjohn (Lily Parradell’s mother): DuckTD\nGladys (Lily’s parlourmaid): Eva Davis\nMaud (Lily’s maid): Leanne Yau\nStage Directions: Larry Wilson\nEdited by: ToddHW",
  "dcterms:subject": ["Arts", "Literature"],
  "author": "LibriVox",
  "conformsTo": "https://www.w3/org/TR/audiobooks/",
  "id": "https://librivox.org/the-mind-the-paint-girl-by-arthur-wing-pinero/",
  "name": "\"Mind The Paint\" Girl, The by PINERO, Arthur Wing",
  "publisher": "",
  "readingOrder": [
    {
      "encodingFormat": "audio/mpeg",
      "name": "Act 1",
      "url": "mindthepaint_1_pinero_64kb.mp3"
    },
    {
      "encodingFormat": "audio/mpeg",
      "name": "Act 2",
      "url": "mindthepaint_2_pinero_64kb.mp3"
    },
    {
      "encodingFormat": "audio/mpeg",
      "name": "Act 3",
      "url": "mindthepaint_3_pinero_64kb.mp3"
    },
    {
      "encodingFormat": "audio/mpeg",
      "name": "Act 4",
      "url": "mindthepaint_4_pinero_64kb.mp3"
    }
  ],
  "resources": [
    {
      "encodingFormat": "image/jpeg",
      "rel": "cover",
      "url": "mind_the_paint_1911.jpg"
    }
  ],
  "type": "Audiobook",
  "url": "https://librivox.org/rss/13939"
}

```

**Comment by danielweck** [comment]
Yes, the duration property on reading order resources sometimes is missing, as well as duration metadata. Counter example (correct / working `publication.json` from W3C LPF audiobook):

```json
{
  "@context": ["https://schema.org", "https://www.w3.org/ns/pub-context"],
  "dcterms:description": "In LibriVox’s Multilingual Poetry Collection, LibriVox volunteers read their favourite public-domain poems in languages other than English. (Summary by David Barnes).",
  "dcterms:subject": ["Arts", "Literature"],
  "author": "LibriVox",
  "conformsTo": "https://www.w3/org/TR/audiobooks/",
  "id": "https://librivox.org/multilingual-poetry-collection-015/",
  "name": "Multilingual Poetry Collection 015 by VARIOUS",
  "publisher": "",
  "readingOrder": [
    {
      "duration": "PT00H00M58S",
      "encodingFormat": "audio/mpeg",
      "name": "Epigram 64",
      "url": "ancientgreek_epigram64_asclepiades_zd_64kb.mp3"
    },
    {
      "duration": "PT00H00M59S",
      "encodingFormat": "audio/mpeg",
      "name": "Bezpomoshtna",
      "url": "bulgarian_bezpomoshtna_hristov_td_64kb.mp3"
    },
    {
      "duration": "PT00H01M33S",
      "encodingFormat": "audio/mpeg",
      "name": "Vora el barranc dels Algadins",
      "url": "catalan_voraelbarranc_llorente_mlb_64kb.mp3"
    },
    {
      "duration": "PT00H01M49S",
      "encodingFormat": "audio/mpeg",
      "name": "Mia Penso",
      "url": "esperanto_miapenso_zamenhof_njb_64kb.mp3"
    },
    {
      "duration": "PT00H00M56S",
      "encodingFormat": "audio/mpeg",
      "name": "A ce jour de Saint Valentin",
      "url": "french_acejour_orleans_sb_64kb.mp3"
    },
    {
      "duration": "PT00H00M46S",
      "encodingFormat": "audio/mpeg",
      "name": "Le temps a laissié son manteau",
      "url": "french_letempsalaissie_orleans_sb_64kb.mp3"
    },
    {
      "duration": "PT00H02M06S",
      "encodingFormat": "audio/mpeg",
      "name": "Himnusz",
      "url": "hungarian_himnusz_kolcsey_dii_64kb.mp3"
    },
    {
      "duration": "PT00H01M52S",
      "encodingFormat": "audio/mpeg",
      "name": "Szózat",
      "url": "hungarian_szozat_vorosmarty_dii_64kb.mp3"
    },
    {
      "duration": "PT00H02M53S",
      "encodingFormat": "audio/mpeg",
      "name": "Il Fiume",
      "url": "italian_ilfiume_dannunzio_dgd_64kb.mp3"
    },
    {
      "duration": "PT00H04M28S",
      "encodingFormat": "audio/mpeg",
      "name": "Le Quattro Stagioni",
      "url": "italian_lequatrostagioni_vivaldi_ghs_64kb.mp3"
    },
    {
      "duration": "PT00H01M04S",
      "encodingFormat": "audio/mpeg",
      "name": "Pianto Antico",
      "url": "italian_piantoantico_carducci_mc_64kb.mp3"
    },
    {
      "duration": "PT00H03M51S",
      "encodingFormat": "audio/mpeg",
      "name": "Elegia I. 1.",
      "url": "latin_elegia1-1_propertius_cm_64kb.mp3"
    },
    {
      "duration": "PT00H01M15S",
      "encodingFormat": "audio/mpeg",
      "name": "Med en vandlilje",
      "url": "norwegian_medenvandlilje_ibsen_ajs_64kb.mp3"
    },
    {
      "duration": "PT00H10M50S",
      "encodingFormat": "audio/mpeg",
      "name": "À morte de Leandro e Hero",
      "url": "portuguese_amortedeleandroehero_bocage_lrl_64kb.mp3"
    },
    {
      "duration": "PT00H01M55S",
      "encodingFormat": "audio/mpeg",
      "name": "Glosa 4",
      "url": "portuguese_glosa4_silva_lrl_64kb.mp3"
    },
    {
      "duration": "PT00H01M20S",
      "encodingFormat": "audio/mpeg",
      "name": "Sfarsitul iernii",
      "url": "romanian_sfarsituliernii_alecsandri_am_64kb.mp3"
    },
    {
      "duration": "PT00H01M33S",
      "encodingFormat": "audio/mpeg",
      "name": "Tunetul",
      "url": "romanian_tunetul_alecsandri_sa_64kb.mp3"
    },
    {
      "duration": "PT00H01M00S",
      "encodingFormat": "audio/mpeg",
      "name": "Daba sustento a un pajarillo un dia",
      "url": "spanish_dabasustento_vega_mm_64kb.mp3"
    },
    {
      "duration": "PT00H02M53S",
      "encodingFormat": "audio/mpeg",
      "name": "Nocturno III. Una noche",
      "url": "spanish_nocturno_silva_edv_64kb.mp3"
    },
    {
      "duration": "PT00H01M00S",
      "encodingFormat": "audio/mpeg",
      "name": "Que tengo yo que mi amistad procuras",
      "url": "spanish_quetengoyo_vega_mm_64kb.mp3"
    }
  ],
  "resources": [
    {
      "encodingFormat": "image/jpeg",
      "rel": "cover",
      "url": "Multilingual_Poetry_Collection_015_1201.jpg"
    }
  ],
  "type": "Audiobook",
  "url": "https://librivox.org/rss/4012"
}
```

**Comment by danielweck** [comment]
Fixed via 55a1129b21c216146289ee1dcb08522ef2290351
